Patches on voidsuits can now be removed

zkxs opened the pull request:
:cl:
bugfix: Voidsuit patches no longer prevent actual repairs
/:cl:

With this, those annoying patches can finally be removed from voidsuits, allowing a savvy repair person to restore a suit to factory new condition. Previously, the only way a patch could be removed was by attacking the suit's wearer.

I've checked to ensure there aren't any other wirecutter `attackby` interactions that would be broken as a result of this, and everything looks good.

I have not validated this in a local server (yet).

<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You'll find a README and example file in .\html\changelogs\ for further instructions.

You can also find a template for adding your changelog directly to the PR description here: https://github.com/Baystation12/Baystation12/wiki/Automatic-changelog-generation
-->

zkxs committed:
Patches on voidsuits can now be removed

:cl:
rscadd: Voidsuit patches can now be removed with wirecutters
/:cl:

With this, those annoying patches can finally be removed from voidsuits, allowing a savvy repair person to restore a suit to factory new condition. Previously, the only way a patch could be removed was by attacking the suit's wearer.

I've checked to ensure there aren't any other wirecutter `attackby` interactions that would be broken as a result of this, and everything looks good.

I have not validated this in a local server (yet).

SierraKomodo commented on code/modules/clothing/spacesuits/breaches.dm:
```diff
@@ -245,6 +245,24 @@
+		if(istype(src.loc, /mob/living/carbon/human))
+			var/mob/living/carbon/human/H = src.loc
+			if(H.wear_suit == src)
+				to_chat(user, "<span class='warning'>You cannot remove patches from \the [src] while it is being worn.</span>")
```
```suggestion
				to_chat(user, SPAN_WARNING("You cannot remove patches from \the [src] while it is being worn."))
```

SierraKomodo commented on code/modules/clothing/spacesuits/breaches.dm:
```diff
@@ -245,6 +245,24 @@
 
+	else if (istype(W, /obj/item/weapon/wirecutters))
+
+		if(istype(src.loc, /mob/living/carbon/human))
```
`src.` is redundant

```suggestion
		if(istype(loc, /mob/living/carbon/human))
```

SierraKomodo commented on code/modules/clothing/spacesuits/breaches.dm:
```diff
@@ -245,6 +245,24 @@
+	else if (istype(W, /obj/item/weapon/wirecutters))
+
+		if(istype(src.loc, /mob/living/carbon/human))
+			var/mob/living/carbon/human/H = src.loc
```
```suggestion
			var/mob/living/carbon/human/H = loc
```

SierraKomodo commented on code/modules/clothing/spacesuits/breaches.dm:
```diff
@@ -245,6 +245,24 @@
+		// remove one patch, chosen arbitrarily
+		for(var/datum/breach/B in breaches)
+			if(B.patched)
+				user.visible_message("<b>[user]</b> uses \the [W] to remove a patch from \the [src].")
```
- Interaction emotes generally don't embolden the user's name
- Always prefix variables with `\a` or `\the` as relevant. 'The monkey uses' vs 'Monkey uses'
- Not a hard requirement but I'd also suggest using span wrappers for messages to make them more distinct from normal text/emotes
- Also not a hard requirement, but visible message accepts a second parameter for a line directed to the user

```suggestion
				user.visible_message(
					SPAN_NOTICE("\The [user] uses \the [W] to remove a patch from \the [src]."),
					SPAN_NOTICE("You use \the [W] to remove a patch from \the [src].")
				)
```

SierraKomodo commented on code/modules/clothing/spacesuits/breaches.dm:
```diff
@@ -245,6 +245,24 @@
+				B.patched = FALSE
+				return
+
+		to_chat(user, "There are no patched breaches to remove with \the [W].")
```
Also not a hard requirement but a suggestion - WARNING shows up in a light red, making it obvious this is a 'It didn't work' message.

```suggestion
		to_chat(user, SPAN_WARNING("There are no patched breaches to remove with \the [W]."))
```

zkxs commented:
@SierraKomodo Thanks for all the feedback! I believe I've addressed all your suggestions.

For the sake of consistency I've made the chat log changes across the entire file, not just the code touched by the new feature.

Techhead0 commented:
The intent when I made patches was that patches would not prevent proper repairs, if that's not working then I'd recommend fixing that. From what I remember, total `damage` is reduced by patching but `burn_damage`/`brute_damage` is not, for this exact reason

zkxs commented:
@Techhead0 Ah, I see. If that was the intent all along then I don't think my proposed wirecutter functionality contributes any value, so I've removed it and fixed the `|| !damage` check on line 52. I've also updated the changelog to reflect the current state of the PR.

So in the end this PR has become a one-liner with a bunch of span style updates tacked on... thus is life.
